### PR TITLE
[MIG] base_fontawesome: Migration to 11.0

### PR DESCRIPTION
**Pull request by etobella**
This module upgrades the fontawesome font to 5.3.1.
It allows to use new icons.
Following the instructions of OCA/web#1042

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
Got this on runbot when logging in, no idea if it is due this module:
image

**Comment by etobella** [comment]
The gitignore was removing the lib folder. I removed it, following the configuration of OCA/web

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Functional review. Works great!

File: base_fontawesome/README.rst
Comment (JordiBForgeFlow) [inline base_fontawesome/README.rst]: brand, not branch

**Review by LoisRForgeFlow — APPROVED** [review]
Tested on runbot, looks great!

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional test

**Comment by JordiBForgeFlow** [comment]
@simahawk we can wait for your final review, as you authored the module in v10, and then this one can be merged.

File: base_fontawesome/templates/assets.xml
Comment (simahawk) [inline base_fontawesome/templates/assets.xml]: can we get rid of unused resources? :)

File: base_fontawesome/templates/assets.xml
Comment (simahawk) [inline base_fontawesome/templates/assets.xml]: AFAIK you should use assets_common to match both back and frontend

File: base_fontawesome/__manifest__.py
Comment (simahawk) [inline base_fontawesome/__manifest__.py]: module version, as per README, should match the version of FA ;)

**Review by simahawk — CHANGES_REQUESTED** [review]

File: base_fontawesome/templates/assets.xml
Comment (etobella) [reply base_fontawesome/templates/assets.xml]: If I replace the elements it is giving problems, related to the usage of FontAwesome font directly on some code by odoo. I will try to review if it is possible to fix the problem.

**Comment by etobella** [comment]
@simahawk I finally applied all the changes that you suggested. There was an issue with the configuration of some buttons, but it seems right now. I also changed the version.

Just in order to leave it prepared for merge, I rebased and squashed all the fixup comits.

File: base_fontawesome/static/lib/fontawesome-5.3.1/css/brands.css
Comment (simahawk) [inline base_fontawesome/static/lib/fontawesome-5.3.1/css/brands.css]: already mentioned in the old PR: do we really have to keep both compressed and uncompressed?
Our repos are already heavy... I think it's better to drop useless files especially if big :)

File: base_fontawesome/templates/assets.xml
Comment (simahawk) [inline base_fontawesome/templates/assets.xml]: can you check indentation? seems messed up :/

File: base_fontawesome/static/lib/fontawesome-5.3.1/css/brands.css
Comment (etobella) [reply base_fontawesome/static/lib/fontawesome-5.3.1/css/brands.css]: I will remove all the unused files.

**Review by simahawk — APPROVED** [review]
LG thanks!

**Comment by simahawk** [comment]
you can squash & merge sir :)

**Comment by etobella** [comment]
Well, I found an small issue on sortable columns. I will fix it first :wink:

**Comment by etobella** [comment]
imagen
Sorting icons are not shown correctly, because It tries to find FontAwesome font but it no longer exists. 
We cannot replace fontawesome original css...

**Comment by simahawk** [comment]
loading them all does not sound like a good idea, sorry :)
Who's causing the issue? Who tries to find what?
Where is the line adding that action? Can't we make it use new FA?

**Comment by etobella** [comment]
The problem is that the old font is referenced on some lessc files:
https://github.com/odoo/odoo/search?l=Less&q=FontAwesome
As they are using it directly it gives problems.

**Comment by simahawk** [comment]
ahh they've changed the name....... `font-family: "Font Awesome 5 Free";  /* updated font-family */`

**Comment by simahawk** [comment]
@etobella do you think we can simply re-define the font-face w/ the old name?
Like here https://github.com/OCA/server-tools/pull/1361/files#diff-5163cade13025e35cf7b4df7687f8697R3787
We could split it to a different compat-only file

**Comment by etobella** [comment]
I finally made it!!
imagen
The problem was related to the link of the font :)

**Pull request merged**
